polar-core/src/partial/constraints.rs

```diff
@@ -97,6 +97,40 @@ impl Constraints {
         constraint_check
     }
 
+    /// Add a constraint that this must contain some known value.
+    ///
+    /// From OTHER in THIS where other is not a partial or variable.
+    pub fn contains(&mut self, other: Term) {
+        // TODO for now, this is represented as an in operation, but we may
+        // want some other representation eventually.
+        // TODO what about non-ground compound terms like [x, 1] in THIS
```

Comment:
Why would we treat `[x, 1] in THIS` any differently from other non-ground terms?

Reply:
Discussed offline - the reason is translation is different on the adapter side, so I want this to be a separate concept.  The representation of constraints as operations is potentially only temporary, and once we have a better concept we can encode this as a separate constraint type which will make writing adaptors easier.

The downside of having a different type for constraints is translating back into a Polar expression would be difficult.  But, the VM can't evaluate the expressions we work with after simplification as is, so I don't think this is much different than our current state.